test/extended/dr/static_pod_rollouts.go

```diff
@@ -0,0 +1,145 @@
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.GinkgoT().Log("ensuring the EtcdCertSignerController has Degraded")
+		status := findCondition(etcdCluster.Status.Conditions, "EtcdCertSignerControllerDegraded")
+		o.Expect(string(status)).To(o.Equal(string(metav1.ConditionTrue)), "expected the EtcdCertSignerController to be degraded due to insufficient quorum")
+
+		g.GinkgoT().Log("ensuring the EtcdEndPointsController has Degraded")
+		status = findCondition(etcdCluster.Status.Conditions, "EtcdEndpointsDegraded")
+		o.Expect(string(status)).To(o.Equal(string(metav1.ConditionTrue)), "expected the EtcdEndPointsController to be degraded due to insufficient quorum")
+
+		// g.GinkgoT().Log("ensuring the TargetConfigController has Degraded")
+		// status = findCondition(etcdCluster.Status.Conditions, "TargetConfigControllerDegraded")
+		// o.Expect(string(status)).To(o.Equal(string(metav1.ConditionTrue)), "expected the TargetConfigController to be degraded due to insufficient quorum")
+
+		// getting the count of installer pods before triggering a rollout
+		installerPodCountPreRolloutTrigger := getPodCount(oc, etcdNamespace, "app=installer")
+
+		g.GinkgoT().Log("setting the hardware speed to Slower to trigger a revision rollout")
+		data := fmt.Sprintf(`{"spec": {"controlPlaneHardwareSpeed": "Slower"}}`)
+		_, err = oc.AdminOperatorClient().OperatorV1().Etcds().Patch(ctx, "cluster", types.MergePatchType, []byte(data), metav1.PatchOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		// allow some time after triggering a rollout before observing the changes in the cluster
+		time.Sleep(60 * time.Second)
+
+		g.GinkgoT().Log("ensuring that triggering a rollout doesn't create any new installer pods due to insufficient quorum")
+		installerPodCountPostRolloutTrigger := getPodCount(oc, etcdNamespace, "app=installer")
+		o.Expect(installerPodCountPreRolloutTrigger).To(o.Equal(installerPodCountPostRolloutTrigger), "triggering a rollout should not create any new installer pods due to insufficient quorum")
+
+	})
+})
+
+// returns the number of static pods in the given namespace that match the given label selector
+func getPodCount(oc *exutil.CLI, namespace, podLabel string) int {
+	pods, err := oc.AdminKubeClient().CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{LabelSelector: podLabel})
+	o.Expect(err).NotTo(o.HaveOccurred())
+	return len(pods.Items)
+}
+
+// returns whether a pod is ready or not
+func isPodReady(pod corev1.Pod) bool {
+	for _, c := range pod.Status.Conditions {
+		if c.Type == corev1.PodReady {
+			return c.Status == corev1.ConditionTrue
+		}
+	}
+	return false
+}
+
+// finds the status of a given condition type from a given list of conditions
+func findCondition(operatorStatusCondition []operatorv1.OperatorCondition, conditionType string) operatorv1.ConditionStatus {
+	for _, condition := range operatorStatusCondition {
+		if condition.Type == conditionType {
+			return condition.Status
+		}
+	}
+	return ""
+}
```

Comment:
I don't have it handy, but there are also likely some helpers for that (I believe in library-go). Otherwise, in latest go the `slices` packages might also be helpful